### ISSUE TITLE
Installing NPM Package do not work inside Vagrant Ubuntu 18.04 LTS box

### ISSUE THREAD

**Issue by jason-migz** [hashicorp/vagrant#11842]
### Vagrant version
Vagrant 2.2.7

### Host operating system

Ubuntu 18.04 LTS

### Guest operating system

Ubuntu 18.04 LTS

### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
    config.vm.box = "base"
  
    config.vm.define "bionic_node_deno_v1" do |bionic_node_deno_v1|
      bionic_node_deno_v1.vm.hostname = "192.168.22.33"
      bionic_node_deno_v1.vm.box = "bento/ubuntu-18.04"
      bionic_node_deno_v1.vm.network :private_network, ip: "192.168.22.33"
      bionic_node_deno_v1.vm.provider "virtualbox" do |bionicv|
        bionicv.memory = 4096
        bionicv.cpus = 4
      end
      bionic_node_deno_v1.vm.synced_folder "~/Documents/vagrant-sync-folder/node-deno-version-1-sf","/home/vagrant/Code"
    end

    config.vm.network "forwarded_port", guest: 8085, host: 8787, host_ip: "127.0.0.1"

  end
```
### Debug output
```
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/vagrant/Code/node_modules/chalk/node_modules/has-flag/package.json.2851319490
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/vagrant/Code/node_modules/chalk/node_modules/has-flag/package.json.2851319490'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/vagrant/.npm/_logs/2020-08-23T07_33_33_057Z-debug.log
```

### Expected behavior

NPM package will successfully installed

### Actual behavior

Got error when try to install the npm package

### Steps to reproduce

1.run vagrant up
2.run vagrant ssh
3.run cd Code
4.run *npm install --save-dev nodemon*
5.tried to run *npm install --save-dev --no-bin-links nodemon* but i still got error

![no-bin-links-error](https://user-images.githubusercontent.com/21366028/90979712-90bff180-e589-11ea-9182-138741257faa.png)
![nodemon-error](https://user-images.githubusercontent.com/21366028/90979713-9289b500-e589-11ea-9771-2f24fd75c006.png)

**Comment by soapy1** [comment]
It looks like this issue is maybe related to this virtualbox issue https://www.virtualbox.org/ticket/19545. By default Vagrant will enable symlinks for it's virtualbox shared folders. This can be verified using the command `VBoxManage getextradata  <vm name> enumerate`. 
I found I was able to get around this issue by binding the node_module directory to a location outside the synced folder using ` sudo mount --bind ~/vagrant_node_modules /Code/node_modules`.
